Packet++/src/Layer.cpp

```diff
@@ -78,6 +84,19 @@ namespace pcpp
 			return true;
 		}
 
+		if (m_Data - m_Packet->m_RawPacket->getRawData() + static_cast<ptrdiff_t>(offsetInLayer) >
+		    static_cast<ptrdiff_t>(m_Packet->m_RawPacket->getRawDataLen()))
+		{
+			PCPP_LOG_ERROR("Requested offset is larger than total packet length");
+			return false;
+		}
+
+		if (m_NextLayer != nullptr && static_cast<ptrdiff_t>(offsetInLayer) > m_NextLayer->m_Data - m_Data)
+		{
+			PCPP_LOG_ERROR("Requested offset exceeds current layer's boundary");
+			return false;
+		}
```

Comment:
Why are these checks needed? All we need to check is that `offsetInLayer` is within the layer's boundary, which means it is smaller than or equal to `m_DataLen`. Why do we need to compare it to the whole packet or involve `m_NextLayer`?